From a7e29f251afc9c341a288252d45fb0150bb2204d Mon Sep 17 00:00:00 2001
From: Chris Morgan <macromorgan@hotmail.com>
Date: Wed, 30 Oct 2024 17:40:28 -0500
Subject: [PATCH] arm64: dts: rockchip: Add idle-states for rk356x

Add the idle-states for the rk356x from the BSP kernel. I observed idle
power draw drop around 0.2w with this change in place on my Anbernic
RG353P device based around the rk3566 SoC.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 arch/arm64/boot/dts/rockchip/rk356x.dtsi | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk356x.dtsi b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
index 938df18a81ff..cd08110720f5 100644
--- a/arch/arm64/boot/dts/rockchip/rk356x.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk356x.dtsi
@@ -55,6 +55,7 @@ cpu0: cpu@0 {
 			reg = <0x0 0x0>;
 			clocks = <&scmi_clk 0>;
 			#cooling-cells = <2>;
+			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			i-cache-size = <0x8000>;
@@ -71,6 +72,7 @@ cpu1: cpu@100 {
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x100>;
 			#cooling-cells = <2>;
+			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			i-cache-size = <0x8000>;
@@ -87,6 +89,7 @@ cpu2: cpu@200 {
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x200>;
 			#cooling-cells = <2>;
+			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			i-cache-size = <0x8000>;
@@ -103,6 +106,7 @@ cpu3: cpu@300 {
 			compatible = "arm,cortex-a55";
 			reg = <0x0 0x300>;
 			#cooling-cells = <2>;
+			cpu-idle-states = <&CPU_SLEEP>;
 			enable-method = "psci";
 			operating-points-v2 = <&cpu0_opp_table>;
 			i-cache-size = <0x8000>;
@@ -113,6 +117,18 @@ cpu3: cpu@300 {
 			d-cache-sets = <128>;
 			next-level-cache = <&l3_cache>;
 		};
+
+		idle-states {
+			entry-method = "psci";
+			CPU_SLEEP: cpu-sleep {
+				compatible = "arm,idle-state";
+				local-timer-stop;
+				arm,psci-suspend-param = <0x0010000>;
+				entry-latency-us = <100>;
+				exit-latency-us = <120>;
+				min-residency-us = <1000>;
+			};
+		};
 	};
 
 	/*
-- 
2.34.1

